test/tests/functional/pbs_svr_dyn_res.py

```diff
@@ -40,26 +40,70 @@
 
 
 class TestServerDynRes(TestFunctional):
+
+    def setup_dyn_res(self, resname, restype, resval):
+        """
+        Helper function to setup server dynamic resources
+        """
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'False'})
+
+        for i in resname:
+            attr = {"type": restype[0]}
+            self.server.manager(MGR_CMD_CREATE, RSC, attr, id=i, expect=True)
+            # Add resource to sched_config's 'resources' line
+            self.scheduler.add_resource(i)
+
+        # Add server_dyn_res entry in sched_config
+        if len(resval) > 1:  # Mutliple resources
+            # To create multiple server dynamic resources in sched_config
+            # from PTL, a list containing "resource !<script>" should be
+            # supplied as value to the key 'server_dyn_res' when calling
+            # set_sched_config().
+            # But this workaround works only if sched_config already has a
+            # server_dyn_res entry.
+            # HACK: So adding a single resource first and then the list.
+            # There wouldn't be any duplicate entries though.
+            a = {'server_dyn_res': resval[0]}
+            self.scheduler.set_sched_config(a)
+            a = {'server_dyn_res': resval}
+        else:
+            a = {'server_dyn_res': resval[0]}
+
+        self.scheduler.set_sched_config(a)
+
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'True'})
+
+    def submit_job(self, attrib):
+        """
+        Helper function to setup node and submit job. Returns job id.
+        """
```

Comment:
on second thought we can avoid this function by moving the setting ncpus part to setUp. for some tests when you have to submit multiple jobs more than twice this will not be useful. Also setting up nodes is part of configuration and must go to setUp.